src/main/java/strategy/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,20 @@
+package strategy.operation;
+
+import dao.FruitShopDaoImpl;
+import exception.FruitShopException;
+import model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    private final FruitShopDaoImpl fruitShopDao = new FruitShopDaoImpl();
+
+    @Override
+    public void execute(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int balance = fruitShopDao.getBalanceByFruit(fruit) - transaction.getQuantity();
```

Comment:
```suggestion
        int newBalance = fruitShopDao.getBalanceByFruit(fruit) - transaction.getQuantity();
```
a bit confusing naming